packages/build-utils/src/clone-env.ts

```diff
@@ -0,0 +1,28 @@
+import type { ProcessEnv } from './types';
+
+const { hasOwnProperty } = Object.prototype;
+
+/**
+ * Identical to `Object.assign()`, but ensures that the `PATH` environment
+ * variable is defined even when the variable is spelled `Path`.
+ *
+ * @param {Object} [...envs] Objects and/or `process.env` to merge.
+ * @returns {Object}
+ */
+export function cloneEnv(...envs: (ProcessEnv | undefined)[]): ProcessEnv {
```

Comment:
```suggestion
export function cloneEnv<T>(...envs: (T | undefined)[]): T {
```

Can we make this generic?

Reply:
I don't think generics will benefit here now that I'm using `Env` which is already pretty generic.